### PR TITLE
[processing] B3DM to GLTF algorithm - handle broken files

### PR DESCRIPTION
**Pull request by nyalldawson**
When GLTF content cannot be parsed from the B3DM, just extract the binary contents anyway to a file.

Allows these broken files to be tested in other applications to determine if it's just the GLTF reader library which doesn't handle them, or actually broken content.

### CODE REVIEW

**Comment by nyalldawson** [comment]
(read diff with ignore whitespace -- it makes it much simpler!)

File: src/analysis/processing/qgsalgorithmb3dmtogltf.cpp [hunk 1]
```diff
@@ -98,56 +98,70 @@ QVariantMap QgsB3DMToGltfAlgorithm::processAlgorithm( const QVariantMap &paramet
...
-    const tinygltf::Mesh &mesh = model.meshes[gltfNode.mesh];
-    feedback->pushDebugInfo( QObject::tr( "Found %1 primitives in default scene node [%2]" ).arg( mesh.primitives.size() ).arg( nodeIndex ) );
-  }
-
-  if ( !b3dmContent.rtcCenter.isNull() )
-  {
-    // transfer B3DM RTC center to GLTF CESIUM_RTC extension
-    tinygltf::Value::Object cesiumRtc;
-    cesiumRtc["center"] = tinygltf::Value( tinygltf::Value::Array
+    // if we can't read the GLTF, then just write the original B3DM content to the output file
```
Comment (nirvn) [inline src/analysis/processing/qgsalgorithmb3dmtogltf.cpp:112]: I'd reword this a bit for clarity. I first misunderstood it as meaning writing the original B3DM file as output file (vs. copying the B3DM's GLTF content even though it failed the load check), which perplexed me :)

**Review by nirvn — APPROVED** [review]

File: src/analysis/processing/qgsalgorithmb3dmtogltf.cpp (hunk 1 above)
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmb3dmtogltf.cpp:112]:
```suggestion
    // if we can't read the GLTF, then just write the original GLTF content to the output file
```
